pkg/api/metrics.go

```diff
@@ -62,19 +62,21 @@ func (hs *HTTPServer) QueryMetricsV2(c *models.ReqContext) response.Response {
 }
 
 func (hs *HTTPServer) toJsonStreamingResponse(qdr *backend.QueryDataResponse) response.Response {
-	statusWhenError := http.StatusBadRequest
-	if hs.Features.IsEnabled(featuremgmt.FlagDatasourceQueryMultiStatus) {
-		statusWhenError = http.StatusMultiStatus
-	}
-
-	statusCode := http.StatusOK
-	for _, res := range qdr.Responses {
-		if res.Error != nil {
-			statusCode = statusWhenError
+	if !hs.Features.IsEnabled(featuremgmt.FlagDatasourceQueryMultiStatus) {
+		statusCode := http.StatusOK
+		for _, res := range qdr.Responses {
+			if res.Error != nil {
+				statusCode = http.StatusBadRequest
+			}
 		}
+		return response.JSONStreaming(statusCode, qdr)
 	}
 
-	return response.JSONStreaming(statusCode, qdr)
+	res := map[string]queryResponse{}
```

Comment:
why create a new one?

Reply:
The idea is that the `featuremgmt.FlagDatasourceQueryMultiStatus` dictates whether or not to use the new response format IE HTTP 207 multi status header with a status field for each each response

Reply:
I just mean use the [QueryDataResponse](https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/data.go#L75) from sdk and put whatever errors you need on it.

Just seems really odd to create and maintain a new structure that can only write, but not read 🤷